crypto-ffi/build.rs

```diff
@@ -31,8 +34,37 @@ fn main() {
     if target_arch == "x86_64" && target_os == "android" {
         let android_home = std::env::var("ANDROID_HOME").expect("ANDROID_HOME not set");
         const ANDROID_NDK_VERSION: &str = "25.2.9519653";
-        const LINUX_X86_64_LIB_DIR: &str = "/toolchains/llvm/prebuilt/linux-x86_64/lib64/clang/14.0.7/lib/linux/";
-        println!("cargo:rustc-link-search={android_home}/ndk/{ANDROID_NDK_VERSION}/{LINUX_X86_64_LIB_DIR}");
+
+        let ndk_root_str = android_home + "/ndk/" + ANDROID_NDK_VERSION;
+        let ndk_root = Path::new(ndk_root_str.as_str());
+        if !ndk_root.exists() {
+            panic!("Error: Couldn't find NDK at path {ndk_root_str}. NDK {ANDROID_NDK_VERSION} is required")
+        }
+
+        let search_path = ndk_root.to_str().unwrap().to_owned() + "/toolchains/llvm/prebuilt/*/lib*/clang/*/lib/linux/";
```

Comment:
Instead of doing the glob search, I suggest we use the output `clang --print-runtime-dir`, which gives the exact directory we need, which is (hopefully) guaranteed to always be correct.

But then the problem becomes, how do we find `clang`?
It's located at `$NDK_ROOT/toolchains/llvm/prebuilt/$ARCH/bin/clang` so maybe we can deduce $ARCH or use some of the information already present under `$NDK_ROOT`?

Reply:
I don't see a way of going from `$NDK_ROOT` to `clang` 😞 

Also, it seems that `clang --print-runtime-dir` doesn't target the linux libs when running on a mac, for example:

```yaml
vitor@Vitors-Laptop bin % ./clang --print-runtime-dir        
/Users/vitor/Library/Android/sdk/ndk/27.0.11718014/toolchains/llvm/prebuilt/darwin-x86_64/lib/clang/18/lib/darwin
```